### PR TITLE
[ClientBuilder] Verify Module has ClientBuilder Class

### PR DESCRIPTION
**Pull request by ijrsvt**
## Why are these changes needed?
* Provides simpler error messages if one `ray.client("...").connect()` is called with a non-existent module or a module that does not have a `ClientBuilder` class!

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/client_builder.py [hunk 1]
```diff
@@ -96,6 +96,8 @@ def _get_builder_from_address(address: Optional[str]) -> ClientBuilder:
         return _LocalClientBuilder(address)
     module_string, inner_address = _split_address(address)
     module = importlib.import_module(module_string)
```
Comment (AmeerHajAli) [inline python/ray/client_builder.py:98]: You might also need to try except this importlib.import_module()

Comment (ijrsvt) [reply python/ray/client_builder.py:98]: Definitely!

**Review by AmeerHajAli — APPROVED** [review]
Do you want to add a pytest check for the exception being raised?

**Comment by ijrsvt** [comment]
> Do you want to add a pytest check for the exception being raised?

Added them now!

**Comment by AmeerHajAli** [comment]
@ijrsvt , this is not a release blocker right?

**Comment by ijrsvt** [comment]
I was thinking no because the ClientBuilder is experimental in this 1.4.0!

**Comment by ijrsvt** [comment]
Windows failures are unrelated (`test_cancel.py`)
OSX Failure also looks unrelated because it times out (likely because CI resources are limited) and is fine because:
1) Fails on master sometimes:
Screen Shot 2021-06-02 at 9 16 01 AM
2) Passes locally: 

```
 ray/tests/test_client_references.py ✓✓✓✓✓✓                                                                                 100% ██████████##vso[results.publish type=JUnit;runTitle='Pytest results';]/Users/ianrodney/Documents/ray/test-output.xml
Skipping uploading of coverage data.

Results (31.64s):
       6 passed
```

**Pull request merged**